src/components/PlaidLink/index.native.tsx

```diff
@@ -1,27 +1,28 @@
 import {useEffect} from 'react';
-import {dismissLink, openLink, useDeepLinkRedirector, usePlaidEmitter} from 'react-native-plaid-link-sdk';
+import {dismissLink, LinkEvent, openLink, useDeepLinkRedirector, usePlaidEmitter} from 'react-native-plaid-link-sdk';
 import Log from '@libs/Log';
 import CONST from '@src/CONST';
-import {plaidLinkDefaultProps, plaidLinkPropTypes} from './plaidLinkPropTypes';
+import PlaidLinkProps from './types';
 
-function PlaidLink(props) {
+function PlaidLink({token, onSuccess = () => {}, onExit = () => {}, onEvent}: PlaidLinkProps) {
     useDeepLinkRedirector();
-    usePlaidEmitter((event) => {
-        Log.info('[PlaidLink] Handled Plaid Event: ', false, event);
-        props.onEvent(event.eventName, event.metadata);
+    usePlaidEmitter((event: LinkEvent) => {
+        Log.info('[PlaidLink] Handled Plaid Event: ', false, event.eventName);
+        onEvent(event.eventName, event.metadata);
     });
     useEffect(() => {
-        props.onEvent(CONST.BANK_ACCOUNT.PLAID.EVENTS_NAME.OPEN, {});
+        onEvent(CONST.BANK_ACCOUNT.PLAID.EVENTS_NAME.OPEN);
```

Comment:
I know that this argument you've removed was probably unnecessary, but please make sure it doesn't cause any regressions.

Reply:
This may actually cause a crash. Please do one of the following:
- Update [AddPlaidBankAccount](https://github.com/Expensify/App/blob/f6ef8b85f94d3f6d7c6c9f90d0c23897f66ce318/src/components/AddPlaidBankAccount.js#L207C3-L207C3) to take into account the possibility of metadata being undefined (cannot access metadata.error_code)
- Revert this change and add a dummy object (+update `types.ts` to mark metadata as required)

Reply:
Done (1st option)